crates/rt/src/thread.rs

```diff
@@ -16,25 +15,30 @@ pub fn wait_for_completion() {
         );
     }
 
-    const MAX_THREADS: usize = 32;
     let cur_thread_id = unsafe { sys::furi_thread_get_current_id() };
     let app_id = unsafe { CStr::from_ptr(sys::furi_thread_get_appid(cur_thread_id)) };
-    let mut thread_ids: [sys::FuriThreadId; MAX_THREADS] = [ptr::null_mut(); MAX_THREADS];
+    let furi_thread_list = unsafe { sys::furi_thread_list_alloc() };
 
     'outer: loop {
-        let thread_count =
-            unsafe { sys::furi_thread_enumerate(thread_ids.as_mut_ptr(), MAX_THREADS as u32) }
-                as usize;
+        let thread_count = unsafe { sys::furi_thread_list_size(furi_thread_list) };
```

Comment:
As I can [see][1], the call to `furi_thread_enumerate` is still required to populate the list of threads.

[1]: https://github.com/flipperdevices/flipperzero-firmware/blob/dev/applications/services/cli/cli_commands.c#L396

Reply:
I missed this one as well :/ 

Sorry I'm giving more work than I'm solving haha. Thanks for taking the time to review :)